### PR TITLE
New insert tag system

### PR DESCRIPTION
**Pull request by ausi**
See #1379

Do we usually link to source code on github for examples?
Should we add more links to such examples here? (currently only the `IfLanguageInsertTag` is linked to).

### CODE REVIEW

**Comment by fritzmg** [comment]
I added a little more description regarding block insert tags and also included a (beautifully) simple (and still super useful) example. wdyt?

I also changed the namespace from `InsertTags` to `InsertTag` as namespaces are usually always singular.

File: docs/dev/framework/insert-tags.md [hunk 1]
```diff
@@ -8,12 +8,210 @@ aliases:
...
+
+The following example will provide an insert tag which transforms a string with
+the `str_rot13` function provided by PHP.
+
+```php
+// src/InsertTag/Rot13InsertTag.php
+namespace App\InsertTag;
+
+use Contao\CoreBundle\DependencyInjection\Attribute\AsInsertTag;
+use Contao\CoreBundle\InsertTag\InsertTagResult;
```
Comment (ausi) [inline docs/dev/framework/insert-tags.md:40]:
```suggestion
use Contao\CoreBundle\InsertTag\Exception\InvalidInsertTagException;
use Contao\CoreBundle\InsertTag\InsertTagResult;
```

File: docs/dev/framework/insert-tags.md [hunk 2]
```diff
@@ -8,12 +8,211 @@ aliases:
...
+
+use Contao\CoreBundle\DependencyInjection\Attribute\AsBlockInsertTag;
+use Contao\CoreBundle\InsertTag\Exception\InvalidInsertTagException;
+use Contao\CoreBundle\InsertTag\ParsedSequence;
+use Contao\CoreBundle\InsertTag\ResolvedInsertTag;
+use Contao\CoreBundle\InsertTag\Resolver\BlockInsertTagResolverNestedResolvedInterface;
+use Contao\CoreBundle\Security\ContaoCorePermissions;
+use Symfony\Bundle\SecurityBundle\Security;
+
+#[AsBlockInsertTag('ifmembergroup', 'endifmembergroup')]
```
Comment (ausi) [inline docs/dev/framework/insert-tags.md]:
```suggestion
#[AsBlockInsertTag('ifmembergroup', endTag: 'endifmembergroup')]
```
Should we add the parameter name to make it more obvious what this refers to?

**Comment by ausi** [comment]
> I added a little more description regarding block insert tags and also included a (beautifully) simple (and still super useful) example. wdyt?

perfect 👍

> I also changed the namespace from `InsertTags` to `InsertTag` as namespaces are usually always singular.

:+1:

File: docs/dev/framework/insert-tags.md [hunk 3]
```diff
@@ -8,12 +8,211 @@ aliases:
...
+in the Contao core bundle: the [`IfLanguageInsertTag`][IfLanguageInsertTag]. It allows you to conditionally output some
+content, depending on the page's language, e.g.:
+
+```
+{{ifnlng::de}}Some English text.{{ifnlng}}
+{{iflng::de}}Some German text.{{iflng}}
+```
+
+When implementing a block insert tag your function receives the wrapped content as a `ParsedSequence` object in addition
+to your `InsertTag`. Your function is then expected to return a `ParsedSequence` again - which could be empty in
```
Comment (ausi) [inline docs/dev/framework/insert-tags.md]:
```suggestion
to your `InsertTag`. The `ParsedSequence` object consists of strings and other parsed (but not yet resolved)
insert tags. Your function is then expected to return a `ParsedSequence` again - which could be empty in
```

**Review by fritzmg — APPROVED** [review]

**Comment by fritzmg** [comment]
Thank you @ausi :)

**Pull request merged**